### PR TITLE
feat(UChild): add UChildEIP3009

### PR DESCRIPTION
**Pull request by pyk**
@dev1644 based on your review I've created new pull request.

This one should work:

Screenshot 2023-08-22 at 04 22 07

### CODE REVIEW

**Comment by pyk** [comment]
@dev1644 command to run the test:

```
 forge test --match-contract UChildEIP3009Test
```

File: contracts/child/ChildToken/UpgradeableChildERC20/UChildEIP3009.sol [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        ));
+        bytes32 digest = toTypedMessageHash(authorization);
+        address signer = ecrecover(digest, v, r, s);
+        require(signer == from, SIGNATURE_INVALID);
+
+        _eip3009_states[from][nonce] = true;
+        emit AuthorizationUsed(from, nonce);
+    }
+
+    /// @dev EIP3009: Check wethere nonce is used or not
```
Comment (dev1644) [inline contracts/child/ChildToken/UpgradeableChildERC20/UChildEIP3009.sol:73]: Typo `wethere`

**Review by dev1644 — COMMENTED** [review]
LGTM, requested one edit on a typo.

File: contracts/child/ChildToken/UpgradeableChildERC20/UChildEIP3009.sol (hunk 1 above)
Comment (pyk) [reply contracts/child/ChildToken/UpgradeableChildERC20/UChildEIP3009.sol]: @dev1644 fixed 👍

**Review by dev1644 — APPROVED** [review]

**Review by BertsLifee — APPROVED** [review]

**Comment by oneski** [comment]
@dev1644 + team can we merge into main repo?

**Comment by dev1644** [comment]
@oneski It's approved from my side, I think we should also request a review from a dev team or at least notify them before merging it.

**Review by DhairyaSethi — APPROVED** [review]

**Pull request merged**